rtt_rosparam/include/rtt_rosparam/RosParamDataSource.hpp

```diff
@@ -0,0 +1,221 @@
+/***************************************************************************
+  tag: Sergio Portoles  Thu Aug 06 09:10:00 CEST 2020  RosParamDataSource.gpp
+
+                        RosParamDataSource.gpp -  description
+                           -------------------
+    begin                : Thu Aug 06 2020
+    copyright            : (C) 2020 Peter Soetens
+    email                : peter.soetens@fmtc.be
+
+ ***************************************************************************
+ *   This library is free software; you can redistribute it and/or         *
+ *   modify it under the terms of the GNU General Public                   *
+ *   License as published by the Free Software Foundation;                 *
+ *   version 2 of the License.                                             *
+ *                                                                         *
+ *   As a special exception, you may use this file as part of a free       *
+ *   software library without restriction.  Specifically, if other files   *
+ *   instantiate templates or use macros or inline functions from this     *
+ *   file, or you compile this file and link it with other files to        *
+ *   produce an executable, this file does not by itself cause the         *
+ *   resulting executable to be covered by the GNU General Public          *
+ *   License.  This exception does not however invalidate any other        *
+ *   reasons why the executable file might be covered by the GNU General   *
+ *   Public License.                                                       *
+ *                                                                         *
+ *   This library is distributed in the hope that it will be useful,       *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU     *
+ *   Lesser General Public License for more details.                       *
+ *                                                                         *
+ *   You should have received a copy of the GNU General Public             *
+ *   License along with this library; if not, write to the Free Software   *
+ *   Foundation, Inc., 59 Temple Place,                                    *
+ *   Suite 330, Boston, MA  02111-1307  USA                                *
+ *                                                                         *
+ ***************************************************************************/
+
+
+#ifndef __RTT_ROSPARAM__ROSPARAMDATASOURCE_HPP
+#define __RTT_ROSPARAM__ROSPARAMDATASOURCE_HPP
+
+// #include "mystd.hpp"
+#include "rtt/internal/DataSource.hpp"
+#include "rtt/internal/DataSourceTypeInfo.hpp"
+#include "rtt/internal/Reference.hpp"
+
+#include "ros/ros.h"
+#include "ros/param.h"
+#include "ros/names.h"
+
+#include <boost/function.hpp>
+#include <exception>
+
+namespace RTT
+{
```

Comment:
Better not add rosparam as a sub-namespace of `RTT`, and instead follow the default convention in ROS to use the package name `rtt_rosparam` as the C++ namespace, like in [rtt_rosparam/rosparam.h](https://github.com/orocos/rtt_ros_integration/blob/2568773e68378bb41e668e2cffc1bf251f3f3484/rtt_rosparam/include/rtt_rosparam/rosparam.h#L68) and other packages in this repository. So the fully-qualified class name would be `rtt_rosparam::RosParamDataSource<T>`. You would have to also fully qualify RTT types in the header, and can add a `using namespace RTT` directive in the source file.